desci-server/src/controllers/nodes/nodesCover.ts

```diff
@@ -1,94 +1,179 @@
+    if (nodeVersion) {
+      const gatewayUrl = cleanupManifestUrl(nodeVersion.manifestUrl);
+      console.log('gatewayUrl', gatewayUrl, nodeVersion.manifestUrl);
+      manifest = (await axios.get(gatewayUrl)).data;
+    }
+
+    // const prevCover = await prisma.nodeCover.findFirst
 
     await prisma.nodeCover.upsert({
-      where: { nodeUuid: nodeUUID + '.' },
-      update: { url: data.url, cid },
-      create: { url: data.url, nodeUuid: nodeUUID + '.', cid },
+      where: { nodeUuid_version: { nodeUuid: nodeUUID + '.', version: version || defaultVersion } },
+      create: {
+        url: url,
+        nodeUuid: nodeUUID + '.',
+        cid,
+        version: version || defaultVersion,
+        name: manifest?.title || node.title,
+      },
+      update: {
+        url: url,
+        cid,
+        name: manifest?.title || node.title,
+      },
     });
 
-    res.send({ ok: true, url: data.url });
+    res.send({ ok: true, url: url, title: manifest?.title || node.title });
   } catch (e) {
     console.log('error', e);
     res.status(404).send({ ok: false, message: e.message || 'Error generating cover image' });
   }
 };
 
-export const setCoverImage = async (req: Request, res: Response, next: NextFunction) => {
+// export const setCoverImage = async (req: Request, res: Response, next: NextFunction) => {
+//   try {
+//     const cid = req.query.cid as string;
+//     const nodeUUID = req.params.uuid as string;
+//     const versionQuery = req.params.version as string;
+//     console.log('versionQuery ', versionQuery, parseVersion(versionQuery));
+//     const user = (req as any).user as User;
+
+//     if (!cid || !nodeUUID || !versionQuery) throw Error('Invalid CID params or NodeUuid query');
+//     // check cid exists in data refs table
+
+//     const node = await prisma.node.findFirst({ where: { uuid: nodeUUID + '.', ownerId: user.id } });
+
+//     if (!node) throw Error('Node not found');
+
+//     const dataRefExists = await prisma.dataReference.findFirst({ where: { cid, nodeId: node.id } });
+//     console.log('dataRefExists', cid, node.id, nodeUUID);
+//     if (!dataRefExists) throw Error('Unknown CID reference');
+
+//     const data = await (
+//       await axios.post(
+//         `${MEDIA_SERVER_API_URL}/v1/nodes/cover/${cid}`,
+//         {},
+//         {
+//           headers: { 'x-api-key': MEDIA_SERVER_API_KEY },
+//         },
+//       )
+//     ).data;
+
+//     // await prisma.nodeCover.upsert({
+//     //   where: { nodeUuid: nodeUUID + '.' },
+//     //   update: { url: data.url, cid },
+//     //   create: { url: data.url, nodeUuid: nodeUUID + '.', cid },
+//     // });
+
+//     res.send({ ok: true, url: data.url });
+//   } catch (e) {
+//     console.log('error', e);
+//     res.status(404).send({ ok: false, message: e.message || 'Error generating cover image' });
+//   }
+// };
+
+export const resetCoverImage = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
can we safely delete these setCoverImage and resetCoverImage? @shadrach-tayo

Reply:
yes